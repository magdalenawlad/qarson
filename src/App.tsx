import React, { useEffect } from 'react';
import { Main } from './common/components/Main';
import { Offers } from './features/offers/components/Offers';
import { fetchOffers } from './features/offers/offersSlice';
import { useDispatch } from 'react-redux';
import { AppDispatch } from './app/store';
import { Header } from './common/components/Header';
import './styles/App.scss';

const App = () =>  {
  const dispatch = useDispatch<AppDispatch>();
  
  useEffect(() => {
    dispatch(fetchOffers());
  }, []);

  return (
    <div className="App">
      <Header />
      <Main>
        <Offers />
      </Main>
    </div>
  );
}

export default App;
