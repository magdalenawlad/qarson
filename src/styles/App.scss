@use './variables' as v;

.App {
  background-color: v.$background;
  height: 100vh;
  width: 100vw;
  overflow: hidden;
  color: v.$textPrimary;
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: space-between;
  font-size: v.$fontSizeRegular;

  header, main {
    width: 100%;
  }

  header {
    height: 60px;
    max-width: v.$content-width;
    padding: 0 1rem;
    display: flex;
    align-items: center;
    justify-content: center;
  
    .logo {
      display: flex;
      align-items: center;
      height: 100%;
      flex: 1;

      img {
        height: 70%;
      }
    }
  
    .icon {
      position: relative;
      padding-right: 0.3rem;;

      svg {
        color: #1d1f20;
      }

      .counter {
        position: absolute;
        top: -5px;
        right: -5px;
        border-radius: 50%;
        background-color: v.$qarsonPrimary;
        color: v.$textPrimary;
        height: 25px;
        width: 25px;
        display: flex;
        justify-content: center;
        align-items: center;
        font-size: v.$fontSizeSmall;
      }
    }

    @media only screen and (max-width: v.$breakpointMobile) {
      background: v.$textPrimary;

      .icon {
        svg {
          color: v.$background;
        }
      }
    }
  }

  main {
    overflow: auto;
    flex: 1;
    display: flex;
    flex-direction: column;
    align-items: center;

    .main-container {
      max-width: v.$content-width;
      padding: 0 1rem;
      width: 100%;
    }
  }
}
