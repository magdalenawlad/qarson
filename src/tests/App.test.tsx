import { renderWithProviders } from './test-utils';
import App from '../App';

test('renders correctly', () => {
  const { container } = renderWithProviders(<App />);

  expect(container).toMatchSnapshot();
});
