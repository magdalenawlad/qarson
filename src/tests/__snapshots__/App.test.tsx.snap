// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="App"
  >
    <header>
      <div
        class="logo"
      >
        <img
          alt="Qarson"
          src="logo.png"
        />
      </div>
      <div
        class="icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
          data-testid="ShoppingCartSharpIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M17 18c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm0-3 1.1-2h7.45c.75 0 1.41-.41 1.75-1.03L21.7 4H5.21l-.94-2H1v2h2l3.6 7.59L3.62 17H19v-2H7z"
          />
        </svg>
      </div>
    </header>
    <main>
      <div
        class="main-container"
      >
        <div
          class="search"
        >
          <div>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root search-textfield css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                  id="quick-filter-text"
                  type="text"
                  value=""
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="SearchIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                    />
                  </svg>
                </div>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-hdw1oc"
                  >
                    <span
                      class="notranslate"
                    >
                      â€‹
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="search-quick-filters"
          />
        </div>
        <div
          class="offers-container"
        >
          <div>
            Loading...
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;
